src/clj/rems/api/catalogue.clj

```diff
@@ -4,17 +4,20 @@
             [rems.api.util] ; required for route :roles
             [rems.db.catalogue :as catalogue]
             [ring.util.http-response :refer :all]
-            [schema.core :as s]))
+            [schema.core :as s]
+            [rems.config :refer [env]]
+            [rems.api.util/check-roles :as check-roles ]))
 
 (s/defschema GetCatalogueResponse
   [CatalogueItem])
 
 (def catalogue-api
   (context "/catalogue" []
     :tags ["catalogue"]
-
     (GET "/" []
       :summary "Get the catalogue of items for the UI (does not include archived items)"
-      :roles #{:logged-in}
       :return GetCatalogueResponse
-      (ok (catalogue/get-localized-catalogue-items {:archived false})))))
+        (cond 
+          (:catalogue-is-public env) (ok (catalogue/get-localized-catalogue-items {:archived false}))
+          (check-roles)  (ok (catalogue/get-localized-catalogue-items {:archived false}))
+          :else nil ))))
```

Comment:
This would be better written
```
(if (or (:catalogue-is-public env)
        (roles/has-roles? :logged-in))
  (ok (catalogue/get-localized-catalogue-items {:archived false}))
  (throw-forbidden))
```